tests/csm/rest/test_iam_users.py

```diff
@@ -1430,3 +1430,490 @@ def test_37783(self):
+        bucket_name = "iam-user-bucket-" + str(int(time.time()))
+        s3_obj = S3TestLib(access_key=resp1.json()["keys"][0]["access_key"],
+                           secret_key=resp1.json()["keys"][0]["secret_key"])
+        try:
+            status, resp = s3_obj.create_bucket(bucket_name)
+            self.log.info("Printing response %s", resp.json())
+            assert_utils.assert_true(status, resp)
+        except Exception as error:
+            self.log.info("Expected exception received %s", error)
+        test_file = "test-object.txt"
+        file_path_upload = os.path.join(TEST_DATA_FOLDER, test_file)
+        if os.path.exists(file_path_upload):
+            os.remove(file_path_upload)
+        if not os.path.isdir(TEST_DATA_FOLDER):
+            self.log.debug("File path not exists, create a directory")
+            system_utils.execute_cmd(cmd=common_cmd.CMD_MKDIR.format(TEST_DATA_FOLDER))
+        system_utils.create_file(file_path_upload, self.file_size)
+        resp = s3_obj.put_object(bucket_name=bucket_name, object_name=test_file,
+                                         file_path=file_path_upload)
+        self.log.info("Removing uploaded object from a local path.")
+        os.remove(file_path_upload)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Step: Verify get object.")
+        resp = s3_obj.get_object(bucket_name, test_file)
+        assert_utils.assert_true(resp[0], resp)
+        self.log.info("[END]Update request with uid and generate-key")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38090')
+    def test_38090(self):
+        """
+        Update request with uid and other parameters randomly
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] STEP 1: Creating IAM user with basic parameters")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        self.log.info("payload :  %s", payload)
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        assert_utils.assert_true(resp.status_code == HTTPStatus.CREATED, "IAM user creation failed")
+        uid = resp.json()["tenant"] + "$" + payload['uid']
+        self.log.info("STEP 2: Perform get iam users")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 3: Update any random parameters for created user")
+        payload = self.csm_obj.iam_user_patch_random_payload()
+        self.log.info("Random payload is %s:", payload)
+        self.log.info("Deleting parameters which would not be present in response for updation")
+        invalid_params = ["access_key", "secret_key", "generate_key", "key_type"]
```

Comment:
"generate_key", "key_type" are applicable for patch

Reply:
modified